### PR TITLE
First development branch

### LINKED ISSUE #3: is it possible to pass redirect_uri value  as prop?
Hi,

If I want to use social-login on different urls of a website, is it possible to pass redirect_uri as prop?

Regards,
Owais

### PR DESCRIPTION
**Pull request by geosp**
Fixes #3 
First development branch

### CODE REVIEW

**Comment by geosp** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>You should add at least 1 more reviewer to the PR</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+1346 -467)  
Please keep it below 500 net changes</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 90c0ca5abcc540447a786e6d48e822735ac32934
</p>

**Comment by geosp** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+1352 -468)  
Please keep it below 500 net changes</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f6de7721cf8ae8c43d043ef2f4a0cb2bf252d4e1
</p>

File: package.json [hunk 1]
```diff
@@ -53,13 +57,13 @@
...
     "nyc": "^11.3.0",
     "prettier": "^1.7.4",
-    "webpack": "^3.8.1",
-    "react": "^16.8.0"
+    "react": "^16.8.0",
+    "webpack": "^3.8.1"
   },
   "dependencies": {
     "bluebird": "^3.5.1",
+    "danger": "^9.2.10",
```
Comment (daedalus28) [inline package.json]: Danger should be a devDependency

File: package.json [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
   "description": "Simple client side social authentication for React applications.",
   "main": "lib/react-social-auth.js",
   "scripts": {
     "build": "webpack -d",
     "test": "babel-node ./node_modules/mocha/bin/_mocha",
     "fmt": "prettier --write *.js; prettier --write '{src,test}/**/*.js'",
-    "lint": "eslint --fix dangerfile.js src/*.js test/*.js"
+    "lint": "eslint --fix dangerfile.js src/*.js test/*.js",
+    "lint:ci": "npm run lint -- -o lint-results.json -f json",
+    "lint:fix": "npm run lint -- --fix",
```
Comment (daedalus28) [inline package.json]: Why three separate lint commands?

File: package.json (hunk 1 above)
Comment (geosp) [reply package.json]: Thanks

File: package.json (hunk 2 above)
Comment (geosp) [reply package.json]: I followed a similar pattern as done in here: https://github.com/smartprocure/futil-js/blob/master/package.json#L16 except that I tried to streamline it. What would you suggest? I don't know when the `lint:fix` command is used. I assume is used by `duti` at some point.

File: package.json [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
-  "version": "1.1.1",
+  "version": "1.1.2",
   "description": "Simple client side social authentication for React applications.",
   "main": "lib/react-social-auth.js",
   "scripts": {
     "build": "webpack -d",
     "test": "babel-node ./node_modules/mocha/bin/_mocha",
     "fmt": "prettier --write *.js; prettier --write '{src,test}/**/*.js'",
-    "lint": "eslint --fix dangerfile.js src/*.js test/*.js"
+    "lint": "eslint dangerfile.js src/*.js test/*.js",
```
Comment (daedalus28) [inline package.json]: there is no `dangerfile` in the reop

Comment (geosp) [reply package.json]: fixed

**Pull request merged**